dpnp/tests/test_fft.py

```diff
@@ -617,81 +567,70 @@ def test_fft_error(self, xp):
 
 
 class TestHfft:
-    def setup_method(self):
-        numpy.random.seed(42)
-
     @pytest.mark.parametrize("dtype", get_all_dtypes(no_none=True))
     @pytest.mark.parametrize("n", [None, 5, 18])
     @pytest.mark.parametrize("norm", [None, "backward", "forward", "ortho"])
-    def test_hfft_1D(self, dtype, n, norm):
+    def test_basic(self, dtype, n, norm):
         x = generate_random_numpy_array(11, dtype, low=-1, high=1)
-        a_np = numpy.sin(x)
+        a = numpy.sin(x)
         if numpy.issubdtype(dtype, numpy.complexfloating):
-            a_np = _make_array_Hermitian(a_np, n)
-        a = dpnp.array(a_np)
+            a = _make_array_Hermitian(a, n)
+        ia = dpnp.array(a)
 
-        result = dpnp.fft.hfft(a, n=n, norm=norm)
-        expected = numpy.fft.hfft(a_np, n=n, norm=norm)
-        # check_only_type_kind=True since numpy always returns float64
+        result = dpnp.fft.hfft(ia, n=n, norm=norm)
+        expected = numpy.fft.hfft(a, n=n, norm=norm)
+        # check_only_type_kind=True since NumPy always returns float64
         # but dpnp return float32 if input is float32
         assert_dtype_allclose(result, expected, check_only_type_kind=True)
 
     @pytest.mark.parametrize(
-        "dtype", get_all_dtypes(no_bool=True, no_none=True, no_complex=True)
+        "dtype", get_all_dtypes(no_none=True, no_complex=True)
     )
     @pytest.mark.parametrize("n", [None, 5, 20])
     @pytest.mark.parametrize("norm", [None, "backward", "forward", "ortho"])
-    def test_ihfft_1D(self, dtype, n, norm):
+    def test_inverse(self, dtype, n, norm):
         x = generate_random_numpy_array(11, dtype, low=-1, high=1)
-        a_np = numpy.sin(x)
-        a = dpnp.array(a_np)
-
-        result = dpnp.fft.ihfft(a, n=n, norm=norm)
-        expected = numpy.fft.ihfft(a_np, n=n, norm=norm)
-        assert_dtype_allclose(result, expected, check_only_type_kind=True)
+        a = numpy.sin(x)  # a.dtype is float16 if x.dtype is bool
+        ia = dpnp.array(a)
 
-    @pytest.mark.parametrize("n", [None, 5, 20])
-    @pytest.mark.parametrize("norm", [None, "backward", "forward", "ortho"])
-    def test_ihfft_bool(self, n, norm):
-        a = dpnp.ones(11, dtype=dpnp.bool)
-        a_np = dpnp.asnumpy(a)
-
-        result = dpnp.fft.ihfft(a, n=n, norm=norm)
-        expected = numpy.fft.ihfft(a_np, n=n, norm=norm)
-        assert_dtype_allclose(result, expected, check_only_type_kind=True)
+        factor = 140 if dtype == dpnp.bool else 8
+        result = dpnp.fft.ihfft(ia, n=n, norm=norm)
+        expected = numpy.fft.ihfft(a, n=n, norm=norm)
+        # check_only_type_kind=True since Intel NumPy always returns complex128
+        assert_dtype_allclose(
+            result, expected, factor=factor, check_only_type_kind=True
+        )
 
-    def test_ihfft_error(self):
+    def test_error(self):
         a = dpnp.ones(11)
         # incorrect norm
         assert_raises(ValueError, dpnp.fft.ihfft, a, norm="backwards")
```

Comment:
Previously we only tested non-inverse functions
```suggestion
        assert_raises(ValueError, dpnp.fft.hfft, a, norm="backwards")
```
Should we align here?

Reply:
In fact, with changes made in https://github.com/IntelPython/dpnp/pull/2393, this test is not needed anymore. Previously was added for improving coverage.

Reply:
I think it makes sense to keep that in case we change the implementation somehow in the future.